test/AllocationsTest.jl

```diff
@@ -0,0 +1,27 @@
+module AllocationsTest
+
+using ForwardDiff
+using BenchmarkTools
+
+include(joinpath(dirname(@__FILE__), "utils.jl"))
+
+@testset "Test seed! allocations" begin
+    x = rand(1000)
+    cfg = ForwardDiff.GradientConfig(nothing, x)
+
+    balloc = @ballocated ForwardDiff.seed!($(cfg.duals), $x, $(cfg.seeds))
```

Comment:
Doesn't just `@allocated` work?

Reply:
I tried, but I wasn't able to interpolate with `@allocated`, so it results in non-zero allocations.

Reply:
Actually, it looks like we can use `@allocated`, we'll just need to duplicate the calls and remove the `getproperty`/`getindex` in the calls:
```julia
@testset "Test seed! allocations" begin
    x = rand(1000)
    cfg = ForwardDiff.GradientConfig(nothing, x)
    duals = cfg.duals
    seeds = cfg.seeds
    seed = cfg.seeds[1]

    balloc = @allocated ForwardDiff.seed!(duals, x, seeds)
    balloc = @allocated ForwardDiff.seed!(duals, x, seeds)
    @test balloc == 0

    balloc = @allocated ForwardDiff.seed!(duals, x, seed)
    balloc = @allocated ForwardDiff.seed!(duals, x, seed)
    @test balloc == 0

    index = 1
    balloc = @allocated ForwardDiff.seed!(duals, x, index, seeds)
    balloc = @allocated ForwardDiff.seed!(duals, x, index, seeds)
    @test balloc == 0

    index = 1
    balloc = @allocated ForwardDiff.seed!(duals, x, index, seed)
    balloc = @allocated ForwardDiff.seed!(duals, x, index, seed)
    @test balloc == 0
end
```
Let me know which one is preferred

Reply:
I think that originally (in e.g. 1.0) julia tried to excluded compilation from the output of `@allocated`, but that proved unreliable, and so that was removed in 1.4 (ish?).
In response to that I added `@ballocated` to `BenchmarkTools` that just pulled the allocations out of the fastest run according to `@benchmark` (which will naturally not be the one that included the timing).

I think `@ballocated` is nicer than duplicating personally, but I will defer to others.

Reply:
I agree that `@ballocated` seems nicer, but I'll also defer to others as this is my first PR to this repo.

Reply:
`@ballocated` doesn't really make sense to me. You run the function thousands of time and then pick out the lowest one. For functions that don't allocate, that will be the second run. So why run the function 998 more times?

Reply:
Ok, I'll switch to the code snippet above

Reply:
Updated.